lib/ansible/modules/files/ini_file.py

```diff
@@ -160,77 +166,140 @@ def do_ini(module, filename, section=None, option=None, value=None,
         ini_lines[-1] += '\n'
         changed = True
 
-    # append a fake section line to simplify the logic
-    ini_lines.append('[')
+    if not section or section == 'null':
+        # Managing options within no section
+        # Regex to look for a section
+        r_section_string = '^\s*\['
```

Comment:
Use raw string notation here and below ([see](https://docs.python.org/3.6/library/re.html#raw-string-notation)).

Reply:
This regex is used with [`re.match`](https://docs.python.org/3.6/library/re.html#search-vs-match), `^` could be removed.